src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java

```diff
@@ -1550,11 +1550,231 @@ public boolean createSitePushToRemote(String siteId, String remoteName, String r
         return toRet;
     }
 
-    public boolean validateRemoteRepositoryConnection(String remoteName, String remoteUrl, String remoteUsername, String remotePassword) {
-        boolean toRet = true;
+    @Override
+    public boolean addRemote(String siteId, String remoteName, String remoteUrl, String authenticationType, String remoteUsername, String remotePassword, String remoteToken, String remotePrivateKey) throws InvalidRemoteUrlException, ServiceException {
+        try {
+            logger.debug("Add remote " + remoteName + " to the sandbox repo for the site " + siteId);
+            Repository repo = helper.getRepository(siteId, SANDBOX);
+            try (Git git = new Git(repo)) {
+                RemoteAddCommand remoteAddCommand = git.remoteAdd();
+                remoteAddCommand.setName(remoteName);
+                remoteAddCommand.setUri(new URIish(remoteUrl));
+                remoteAddCommand.call();
+            } catch (URISyntaxException e) {
+                logger.error("Remote URL is invalid " + remoteUrl, e);
+                throw new InvalidRemoteUrlException();
+            } catch (GitAPIException e) {
+                logger.error("Error while adding remote " + remoteName + " (url: " + remoteUrl + ") for site " + siteId, e);
+                throw new ServiceException("Error while adding remote " + remoteName + " (url: " + remoteUrl + ") for site " + siteId, e);
+            }
+
+            logger.debug("Inserting remote " + remoteName + " for site " + siteId + " into database.");
+            Map<String, String> params = new HashMap<String, String>();
+            params.put("siteId", siteId);
+            params.put("remoteName", remoteName);
+            params.put("remoteUrl", remoteUrl);
+            params.put("authenticationType", authenticationType);
+            params.put("remoteUsername", remoteUsername);
+
+            logger.debug("Generate salt to encrypt password");
+            String salt = RandomStringUtils.randomAlphanumeric(16);
```

Comment:
I see the salt here, how does it relate, if at all, to the salt in the yaml config?

Reply:
random salt from runtime (stored in db) + salt from yaml config is used to encrypt password